src/main/java/com/amazonaws/encryptionsdk/kmsv2/AwsKmsMrkAwareMasterKeyProvider.java

```diff
@@ -112,19 +106,17 @@ public Builder withDiscoveryMrkRegion(
      *
      * <p>Because the regional client supplier fully controls the client construction process, it is
      * not possible to configure the client through methods such as {@link
-     * #withCredentials(AWSCredentialsProvider)} or {@link #withClientBuilder(AWSKMSClientBuilder)};
-     * if you try to use these in combination, an {@link IllegalStateException} will be thrown.
+     * #builderSupplier(Supplier)}; if you try to use these in combination, an {@link
+     * IllegalStateException} will be thrown.
      *
-     * @see
-     *     KmsMasterKeyProvider.Builder#withCustomClientFactory(KmsMasterKeyProvider.RegionalClientSupplier)
+     * @see KmsMasterKeyProvider.Builder#customRegionalClientSupplier(RegionalClientSupplier)
      */
-    public Builder withCustomClientFactory(
-        KmsMasterKeyProvider.RegionalClientSupplier regionalClientSupplier) {
-      if (templateBuilder_ != null) {
+    public Builder customRegionalClientSupplier(RegionalClientSupplier regionalClientSupplier) {
```

Comment:
This might be the right move regardless, but just noting that this name change is one more way that customers could potentially get confused in this migration. Any migration guide should make it clear that anyone previously specifying "customerClientFactory" should now specify "regionalClientSupplier".

Reply:
I agree. As someone trying to figure out what was going on here, I didn't know what part of the supplier was _custom_ and I think for new users, that would hold as well. I will make sure to surface this with June.